Mathlib/RingTheory/DedekindDomain/Ideal.lean

```diff
@@ -1234,6 +1234,12 @@ theorem Ideal.IsPrime.mul_mem_pow (I : Ideal R) [hI : I.IsPrime] {a b : R} {n :
   exact Or.inr (Prime.pow_dvd_of_dvd_mul_right ((Ideal.prime_iff_isPrime hI0).mpr hI) _ ha h)
 #align ideal.is_prime.mul_mem_pow Ideal.IsPrime.mul_mem_pow
 
+theorem Ideal.IsPrime.mem_pow_mul (I : Ideal R) [hI : I.IsPrime] {a b : R} {n : ℕ}
+    (h : a * b ∈ I ^ n) : a ∈ I ^ n ∨ b ∈ I := by
+  rw [mul_comm] at h
+  rw [or_comm]
+  exact Ideal.IsPrime.mul_mem_pow _ h
+
```

Comment:
Is this needed by this PR? It seems out of scope to me.

Reply:
I needed it to make one proof work, it was too complicated to guess how the left/right, prime/unprime should be arranged while writing this lemma as the totally symmetric version of the one just above it was a trivial move.

Reply:
Can you put this one in its own PR?